app/javascript/crayons/Pills/Pill.jsx

```diff
@@ -0,0 +1,93 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { useState } from 'preact/hooks';
+import classNames from 'classnames/bind';
+import { Icon } from '@crayons';
+import XIcon from '@images/x.svg';
+
+export const Pill = ({
+  children,
+  element = 'button',
+  iconLeft,
+  iconRight,
+  iconRightDestructive,
+  className,
+  tooltip,
```

Comment:
I'm wondering what kind of cases are we thinking we'll have pills that aren't interactive elements like buttons and links? In my head these pills look very much interactive, so I'm just a bit unsure about things like `span` and `li` being used with the same look 🤔

Reply:
I'm sure you're currently thinking about the Member Details view from Figma and how these are being used there. But one thing we haven't predicted there (and that fact itself isn't really a disaster) is that sometimes we want to represent similar data but in read-only manner. 

In fact, these role pills can actually be in read-only format when you're looking at your own admin profile, for which you can't remove roles (another admin has to do it).

Even though that's a very specific scenario I think I could find few other examples, where we would want similar possibility.

Just think of it as a pattern to represent set of values - these don't really always have to be editable objects.

Reply:
> I honestly don't think there's a sensible reason to differentiate these from "actionable" pills

It's not a hill I'm going to die on, but I just feel like we've been putting a lot of work into making buttons and links visually distinct, and this component gives not just buttons and links, but non-interactive elements the same UI. How is a user to know what they can click on 🤷‍♀️  

We also can't rely on tooltips for communicating why these items aren't interactive, because keyboard users won't be able to trigger those tooltips.

Reply:
How can we make "Design 🤝 Accessibility" here? :D

I can totally see your concerns here but I'm also trying to make _design feel satisfied_ (if that makes sense?)... The primary goal of pills is to represent some data. The actionable part (making pill trigger do something on click) is I think a bit more "secondary". But at the same time, and for the sake of simplicity, we don't want to introduce clunky UI...

I'll DM you so we can find a good middle ground together.

Reply:
Chatted with @aitchiss and we decided to go fully a11y compliant. Suzanne also told me we can ship non-accessible changes when she's on vacations.